### PR TITLE
search in (too) large result sets

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #7753

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

manually

### CODE REVIEW

**Comment by DaanHoogland** [comment]
This is only a domain specific change handler, as of now. I intend to make the solution a bit more generic.

**Comment by DaanHoogland** [comment]
@blueorangutang ui

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: ui/src/components/view/SearchView.vue [hunk 1]
```diff
@@ -447,9 +461,9 @@ export default {
       this.inputKey = this.fieldValues['tags[0].key'] || null
       this.inputValue = this.fieldValues['tags[0].value'] || null
     },
-    fetchZones () {
+    fetchZones (searchKeyword) {
       return new Promise((resolve, reject) => {
-        api('listZones', { showicon: true }).then(json => {
+        api('listZones', { showicon: true, keyyword: searchKeyword }).then(json => {
```
Comment (Pearl1594) [inline ui/src/components/view/SearchView.vue:466]:
```suggestion
        api('listZones', { showicon: true, keyword: searchKeyword }).then(json => {
```

**Comment by Pearl1594** [comment]
Had a small doubt: In case the result of a list API **with the search keyword passed** returns a list > 500 items, would we still have an issue viewing it in the UI?

**Comment by DaanHoogland** [comment]
> Had a small doubt: In case the result of a list API **with the search keyword passed** returns a list > 500 items, would we still have an issue viewing it in the UI?

yes @Pearl1594 , especially in my tests (i thought I had already commented on this) I create a-z and than subdomains xa-xz and next a 1st, 2nd and 3rd subdomain in all those. Now there are 676 domain named 1st and if you search those you get only the first "few". This is only an issue if the complete name is the same for more than 500 domains. In this case we  need further improvement but I think these is a very extreme corner case and we can do with this solution right now. Good to keep in mind though.
One solution is to let the API search on full path, i.e. 'z/1st' returns all '.*z/1st' child domains of any domain ending in 'z'. I am not certain how complicated that would be, it would be a change on the APIthat is an improvement on UI automagically.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Review by kiranchavala — COMMENTED** [review]
@DaanHoogland can you also fix the search view for the account creation wizard
Screenshot 2023-07-27 at 10 50 32 AM

**Review by kiranchavala — COMMENTED** [review]
@DaanHoogland  

Also for the domain view

Screenshot 2023-07-27 at 10 54 17 AM

**Review by kiranchavala — COMMENTED** [review]
@DaanHoogland  for the create network wizard 

Screenshot 2023-07-27 at 11 03 43 AM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@kiranchavala the search view is a generic component, also the domain overview already worked by typing part of the name (it will never show more than 500 domains )
Can you be more precise in your feedback please?

**Comment by DaanHoogland** [comment]
> @DaanHoogland for the create network wizard
> 
> Screenshot 2023-07-27 at 11 03 43 AM

this one does not look wrong. What is the issue here?

**Comment by kiranchavala** [comment]
@DaanHoogland 

The domain drop down view doesn't list the domains greater than 499

Also the Ui doesn't allow me to select the domains  or manually type in domain id greater than 499

1. Account creation wizard
2. Network creation wizard


3.

**Comment by DaanHoogland** [comment]
> @DaanHoogland
> 
> The domain drop down view doesn't list the domains greater than 499
> 
> Also the Ui doesn't allow me to select the domains or manually type in domain id greater than 499
> 
>     1. Account creation wizard
> 
>     2. Network creation wizard

did you try selecting by first partially enter the name?

**Comment by kiranchavala** [comment]
> @DaanHoogland
> 
> Also for the domain view
> 
> Screenshot 2023-07-27 at 10 54 17 AM

@DaanHoogland 

Also there is no pagesize view for list domains in the UI.

The domain listing stops at 500

**Comment by kiranchavala** [comment]
> > @DaanHoogland
> > The domain drop down view doesn't list the domains greater than 499
> > Also the Ui doesn't allow me to select the domains or manually type in domain id greater than 499
> > ```
> > 1. Account creation wizard
> > 
> > 2. Network creation wizard
> > ```
> 
> did you try selecting by first partially enter the name?

yes it was not listing the domains >500 even if i type partially

Screenshot

**Comment by DaanHoogland** [comment]
> > @DaanHoogland
> > Also for the domain view
> > Screenshot 2023-07-27 at 10 54 17 AM
> 
> @DaanHoogland
> 
> Also there is no pagesize view for list domains in the UI.
> 
> The domain listing stops at 500

no, not if you search for the (partial) name:

image

**Comment by DaanHoogland** [comment]
> > > @DaanHoogland
> > > The domain drop down view doesn't list the domains greater than 499
> > > Also the Ui doesn't allow me to select the domains or manually type in domain id greater than 499
> > > ```
> > > 1. Account creation wizard
> > > 
> > > 2. Network creation wizard
> > > ```
> > 
> > 
> > did you try selecting by first partially enter the name?
> 
> yes it was not listing the domains >500 even if i type partially
> 
> Screenshot

ok, this one I would like to make a separate ticket for as it seems to be a different component (together with the network creation wizard) if you don't mind.

**Review by kiranchavala — APPROVED** [review]
LGTM , tested the feature manually and its working fine

**Review by Pearl1594 — APPROVED** [review]

**Pull request merged**